internal/agent/device/status/status.go

```diff
@@ -81,7 +83,12 @@ func (m *StatusManager) Get(ctx context.Context) *v1alpha1.DeviceStatus {
 	return m.device.Status
 }
 
+func (m *StatusManager) reset() {
+	clear(m.device.Status.Applications.Data)
+}
+
 func (m *StatusManager) Collect(ctx context.Context) error {
+	m.reset()
```

Comment:
should we see if what is collected is equal to what we have before we reset()

Reply:
The applications are written to a map in the status (no secondary map).  If no reset is done before `Collect` then removed applications will never be removed from the map.
In order to do the suggested option, we will need a temporary status and then compare.  It will have to be done in many places (ie conditions, summary).  So this was the simplest solution for now.

Reply:
makes sense